build/test-e2e-go/gke/Dockerfile

```diff
@@ -17,7 +17,7 @@ FROM gcr.io/google.com/cloudsdktool/cloud-sdk:425.0.0-slim as gcloud-install
 RUN apt-get install -y kubectl google-cloud-sdk-gke-gcloud-auth-plugin
 
 # Build e2e image
-FROM golang:1.17.7-alpine as kpt-config-sync-e2e
```

Comment:
Could we pull out this Go bump as a separate change, e.g. in case we need to revert one of these changes?

Reply:
The rest of the code was already updated to 1.19, it's just the e2e image here. It's required to build with the newer libraries. So if we want to revert the Go bump it'll require reverting these lib updates too.

Reply:
But I can pull out the Go image change, if you think it's meaningful.

Reply:
I 'd suggest submitting the Go bump first, but I won't give you a hard time about it